jena-base/src/main/java/org/apache/jena/atlas/logging/LogCtlLog4j2.java

```diff
@@ -21,96 +21,320 @@
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Objects;
 
-import org.apache.jena.atlas.io.IO;
-import org.apache.jena.atlas.lib.StrUtils;
+import org.apache.commons.io.FilenameUtils;
 import org.apache.logging.log4j.Level;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.core.config.Configuration;
 import org.apache.logging.log4j.core.config.ConfigurationFactory;
 import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Configurator;
+import org.apache.logging.log4j.core.config.json.JsonConfigurationFactory;
 import org.apache.logging.log4j.core.config.properties.PropertiesConfigurationFactory;
+import org.apache.logging.log4j.core.config.xml.XmlConfigurationFactory;
+import org.apache.logging.log4j.core.config.yaml.YamlConfigurationFactory;
 
 /**
  * Additional logging control, for Log4j2 as used by jena-cmds.
- * <br/>
- * This class pulls in log4j2.
- * <br/>
- * This class is split out from {@link LogCtl} to decouple the class loading dependencies.
+ * <p>
+ * This class depends on log4j2-api and also log4j2-core.
+ * These are &lt;optional&gt; dependencies for Jena which can use any slf4j provider.
+ * <p>
+ * This class is split out from {@link LogCtl} to decouple the dependencies.
+ * <p>
+ * This class is not used if log4j2 is not used.
+ * All logging initialization
```

Comment:
All logging initialization (….) ? Was reading it but the sentence suddenly ended ? :grimacing:

Reply:
Fixed